docs/code-generation/generate-hcl.md

```diff
@@ -11,43 +11,40 @@ next:
   link: '/code-generation/generate-file'
 ---
 
-# HCL Code Generation
+# Generating HCL Code with Terramate
 
-Terramate supports the generation of arbitrary HCL code referencing
-[Terramate defined data](../data-sharing/index.md).
+## Configuration files and `generate_hcl` blocks
 
-The generated code can then be composed/referenced by any Terraform code
-inside a stack (or any other tool that uses HCL, like [Packer](https://www.packer.io/)).
+To begin generating HCL code with Terramate, developers need to define configuration files that specify the desired code components and their corresponding values. Terramate utilizes the `generate_hcl` block to generate HCL code based on these configurations. Multiple generate_hcl blocks can be defined within a [configuration file](https://terramate.io/docs/cli/configuration/), each responsible for generating a specific code component.
 
-HCL code generation is done using `generate_hcl`
-blocks in [Terramate configuration files](../configuration/index.md).
+## Understanding the `generate_hcl` block structure
 
-The code may include:
+The `generate_hcl` block follows a specific structure, consisting of a label, a file path, and a content block. The label provides a unique identifier for the generated code component, while the file path determines the location where the generated code will be saved. The content block contains the HCL code that will be generated.
 
-* Blocks, sub blocks, etc
-* Attributes initialized by literals
-* Terramate Global references
-* Terramate Metadata references
-* Expressions using interpolation, functions, etc
-* Dynamic blocks using the `tm_dynamic` block type
```

Comment:
Is it already clear that all of this is supported in the `content` block without mentioning them? @mariux 
The `content` block is exceptionally the only block where the user is free to include custom blocks, subblocks etc at will, all the other Terramate blocks have a fixed schema.
Maybe the wording can be changed to make that clear without having to enumerate all of the possible features allowed?

Reply:
yes, this should be kept/made clear...

Reply:
Okay. I will include this.